services/QuillLMS/client/app/bundles/Teacher/components/assignment_flow/create_unit/custom_activity_pack/topic_filters.tsx

```diff
@@ -134,42 +168,53 @@ const TopicToggle = ({filteredActivities, grouping, uniqueLevelTwoTopics, topicF
   )
 }
 
-const TopicFilters = ({ activities, filterActivities, topicFilters, handleTopicFilterChange, }: TopicFiltersProps) => {
-  let levelTwoTopics = []
-  let levelThreeTopics = []
-  activities.forEach(a => {
-    const levelTwoTopic = a.topics && a.topics.filter(t => Number(t.level) === 2)
-    const levelThreeTopic = a.topics && a.topics.filter(t => Number(t.level) === 3)
-    levelTwoTopics = levelTwoTopics.concat(levelTwoTopic)
-    levelThreeTopics = levelThreeTopics.concat(levelThreeTopic)
-  })
-
-  const uniqueLevelTwoTopicIds = Array.from(new Set(levelTwoTopics.map(a => a.id)))
-  const uniqueLevelTwoTopics = uniqueLevelTwoTopicIds.map(id => levelTwoTopics.find(t => t.id === id))
-  const uniqueLevelThreeTopicIds = Array.from(new Set(levelThreeTopics.map(a => a.id)))
-  const uniqueLevelThreeTopics = uniqueLevelThreeTopicIds.map(id => levelThreeTopics.find(t => t.id === id))
+const TopicFilters = ({ activities, filterActivities, topicFilters, handleTopicFilterChange, topics, }: TopicFiltersProps) => {
+  let levelOneTopicsInUse = []
+  let uniqueLevelOneTopics = []
+  let uniqueLevelTwoTopics = []
+  let filteredActivities = []
```

Comment:
Why do we need to instantiate this here? Won't just declaring it as `filterActivities(TOPIC_FILTERS)` return an empty array if that's the case?